pattern_matching.md

```diff
@@ -18,3 +18,97 @@ Note that some CL implementations include regexp facilities, notably
 [CLISP](http://clisp.sourceforge.net/impnotes.html#regexp) and
 [ALLEGRO CL](https://franz.com/support/documentation/current/doc/regexp.htm). If
 in doubt, check your manual or ask your vendor.
+
+Description provided below is far from complete, so don't forget to
+check the reference manual that comes along with the CL-PPCRE library.
+
+
+## Using PPCRE
+
+CL-PPCRE (abbreviation for Portable Perl-compatible regular
+expressions) is a portable regular expression library for Common Lisp
+with a broad set of features and good performance. It has been ported
+to a number of Common Lisp implementations and can be easily installed
+(or added as a dependency) via Quicklisp:
+
+~~~lisp
+(ql:quickload "cl-ppcre")
+~~~
+
+It might be more convenient to use CL-PPCRE with the CL-INTERPOL
+library. CL-INTERPOL is a library for Common Lisp which modifies the
+reader in a way that introduces interpolation within strings similar
+to Perl or Unix Shell scripts.
+
+In addition to loading the CL-INTERPOL library, initialization call
+must be made to properly configure the Lisp reader. This is
+accomplished by either calling the `enable-interpol-syntax` function
+from the REPL or placing that call in the source file before using any
+of its features:
+
+~~~lisp
+(interpol:enable-interpol-syntax)
+~~~
+
+Basic operations with the CL-PPCRE library functions are described
+below.
+
+
+## Looking for matching patterns
+
+The `scan` function tries to match the given pattern and on success
+returns four values - the start of the match, the end of the match,
```

Comment:
I would thinking about rewording this: they come back as multiple-values values, rather than all in a list or vector. :thinking: